derrida/books/management/commands/instance_data.py

```diff
@@ -46,7 +46,17 @@ def handle(self, *args, **kwargs):
         if kwargs['directory']:
             base_filename = os.path.join(kwargs['directory'], base_filename)
 
-        filtered_instances = Instance.objects.filter(Q(cited_in__isnull=False) | Q(reference__isnull=False)).distinct()
+        queried_instances = Instance.objects.filter(Q(cited_in__isnull=False) | Q(reference__isnull=False)).distinct()
+        # Find all the instances that have parent instances and include those parent instances in our dataset
+        instances_with_parents = Instance.objects.filter(collected_in__isnull=False)
+        # Collect all the IDs necessary to perform a filter
+        parent_instance_ids = [instance.collected_in.id for instance in instances_with_parents]
+        filtered_instance_ids = [instance.id for instance in queried_instances]
+        collected_instance_ids = set(parent_instance_ids + filtered_instance_ids)
+        # Filter for those parents and our custom queried instances
+        filtered_instances = Instance.objects.filter(id__in=collected_instance_ids)
```

Comment:
@rlskoeser why do I have a sneaking suspicion you have a more elegant way of doing this 😄 this "works" but I'm looking forward to hearing what you have to say!

Reply:
I half understood what you were talking about in our meeting, but now I actually understand what's going on here 😄 ✅ 37dd3051d1157a886c0b30ca5c09fea4fc2fc5cc